include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -108,92 +115,184 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
         template <class RandomIt>
         Impl(RandomIt breaks_begin, RandomIt breaks_end);
 
+        /// @brief Copy-constructs from another Impl with different Kokkos memory space
+        template <class OriginMemorySpace>
+        explicit Impl(Impl<DDim, OriginMemorySpace> const& impl)
+            : m_domain(impl.m_domain)
+            , m_nknots(impl.m_nknots)
+        {
+        }
+
+        /// @brief Copy-constructs
         Impl(Impl const& x) = default;
 
+        /// @brief Move-constructs
         Impl(Impl&& x) = default;
 
+        /// @brief Destructs
         ~Impl() = default;
 
+        /// @brief Copy-assigns
         Impl& operator=(Impl const& x) = default;
 
+        /// @brief Move-assigns
         Impl& operator=(Impl&& x) = default;
 
+        /** @brief Evaluates BSplines at a given coordinate
+         *
+         * The values are computed for every BSplines at the given coordinate x. A spline approximation at coordinate x is a linear combination of those BSplines evaluations weigthed with splines coefficients of the spline-transformed initial discrete function.
+         * @param[out] values The values of the BSplines evaluated at coordinate x. It has to be a (1+degree) 1D mdspan.
+         * @param[in] x The coordinate where BSplines are evaluated.
+         * @return TODO
+         */
```

Comment:
```suggestion
        /** @brief Evaluates non-zero B-splines at a given coordinate.
         *
         * The values are computed for every B-spline with support at the given coordinate x. There are only (degree+1)
         * B-splines which are non-zero at any given point. It is these B-splines which are evaluated.
         * This can be useful to calculate a spline approximation of a function. A spline approximation at coordinate x
         * is a linear combination of these B-spline evaluations weighted with spline coefficients of the spline-transformed
         * initial discrete function.
         *
         * @param[out] values The values of the B-splines evaluated at coordinate x. It has to be a 1D mdspan with (1+degree) elements.
         * @param[in] x The coordinate where B-splines are evaluated.
         * @return The index of the first B-spline which is evaluated.
         */
```